packages/snap/src/rpc/signMessage.ts

```diff
@@ -0,0 +1,17 @@
+import {Message, SignedMessage, transactionSerialize, transactionSign} from "@zondax/filecoin-signing-tools/js";
+import {Wallet} from "../interfaces";
+import {getKeyPair} from "../filecoin/account";
+import {showConfirmationDialog} from "../util/confirmation";
+
+export async function signMessage(wallet: Wallet, message: Message): Promise<SignedMessage> {
+  const keypair = await getKeyPair(wallet);
+  const tx = transactionSerialize(message);
+  const confirmation = await showConfirmationDialog(
+    wallet,
+    `Do you want to sign message ${tx} with account ${keypair.address}?`
```

Comment:
@mpetrun5 @mpetrunic is this better ?

Reply:
We should display formatted unserialized data. bunch of hex symbols won't help out wheter you are signing data you think you are signing :smile: